src/server/generateServerInjector.js

```diff
@@ -1,25 +1,21 @@
-import titleGenerator from './generators/titleGenerator'
-import attrsGenerator from './generators/attrsGenerator'
-import tagGenerator from './generators/tagGenerator'
+import { titleGenerator, attributeGenerator, tagGenerator } from './generators'
 
-export default function _generateServerInjector (options = {}) {
-  /**
-   * Converts a meta info property to one that can be stringified on the server
-   *
-   * @param  {String} type - the type of data to convert
-   * @param  {(String|Object|Array<Object>)} data - the data value
-   * @return {Object} - the new injector
-   */
-  return function generateServerInjector (type, data) {
-    switch (type) {
-      case 'title':
-        return titleGenerator(options)(type, data)
-      case 'htmlAttrs':
-      case 'bodyAttrs':
-      case 'headAttrs':
-        return attrsGenerator(options)(type, data)
-      default:
-        return tagGenerator(options)(type, data)
-    }
+/**
+ * Converts a meta info property to one that can be stringified on the server
+ *
+ * @param  {String} type - the type of data to convert
+ * @param  {(String|Object|Array<Object>)} data - the data value
+ * @return {Object} - the new injector
+ */
+
+export default function generateServerInjector(options, type, data) {
+  if (type === 'title') {
+    return titleGenerator(options, type, data)
+  }
+
+  if (['htmlAttrs', 'headAttrs', 'bodyAttrs'].includes(type)) {
```

Comment:
Const

Reply:
fyi, there are ~10 cases in Nuxt where also this inline structure is used (did a quick grep on `].includes`)